lib/jsonapi_compliable/adapters/active_record/base.rb

```diff
@@ -2,9 +2,124 @@ module JsonapiCompliable
   module Adapters
     module ActiveRecord
       class Base < ::JsonapiCompliable::Adapters::Abstract
-        # (see Adapters::Abstract#filter)
-        def filter(scope, attribute, value)
-          scope.where(attribute => value)
+        def filter_string_eq(scope, attribute, value, is_not: false)
+          column = scope.klass.arel_table[attribute]
+          clause = column.lower.eq_any(value.map(&:downcase))
+          is_not ? scope.where.not(clause) : scope.where(clause)
```

Comment:
You’re repeating this logic in every type of filter. I think you’re missing an optional adapter method for performing negations.

Reply:
The problem is, not all adapters are going to be capable of "take this clause and not-ify it". This is actually an implementation detail of the AR adapter - the real method adapters need to write is `filter_string_not_eq`, where they can do whatever they want.

Nothing is stopping an adapter implementor from doing what you propose as part of their implementation; I'm not sure if we buy much in this case though.